pymc/tests/test_model.py

```diff
@@ -702,6 +703,67 @@ def test_nested_model_coords():
     assert set(m2.RV_dims) < set(m1.RV_dims)
 
 
+def test_shape_error_from_fixed_dims():
+    """
+    with #5751 dimensions set by add_coord() are default TensorConstant type
```

Comment:
Nit: Change docstrings to match numpy style. Notably first line contains summary, and subsequent notes fit in a paragraph below.

https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard